cli/command/service/opts.go

```diff
@@ -660,6 +666,88 @@ func (options *serviceOptions) makeEnv() ([]string, error) {
 	return currentEnv, nil
 }
 
+func (options *serviceOptions) ToPrivileges(flags *pflag.FlagSet) (*swarm.Privileges, error) {
+	// we're going to go through several possible uses of the Privileges
+	// struct, which may or may not be used. If some stage uses it (after the
+	// first), we'll check if it's nil and create it if it hasn't been created
+	// yet.
+	var privileges *swarm.Privileges
+	if options.credentialSpec.String() != "" && options.credentialSpec.Value() != nil {
+		privileges = &swarm.Privileges{
+			CredentialSpec: options.credentialSpec.Value(),
+		}
+	}
+
+	if flags.Changed(flagNoNewPrivileges) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		privileges.NoNewPrivileges = options.noNewPrivileges
+	}
+
+	if flags.Changed(flagAppArmor) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		switch options.appArmor {
+		case "default":
+			privileges.AppArmor = &swarm.AppArmorOpts{
+				Mode: swarm.AppArmorModeDefault,
+			}
+		case "disabled":
+			privileges.AppArmor = &swarm.AppArmorOpts{
+				Mode: swarm.AppArmorModeDisabled,
+			}
+		default:
+			return nil, errors.Errorf(
+				"unknown AppArmor mode %q. Supported modes are %q and %q",
+				options.appArmor,
+				swarm.AppArmorModeDefault,
+				swarm.AppArmorModeDisabled,
+			)
+		}
+	}
+
+	if flags.Changed(flagSeccomp) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		switch options.seccomp {
+		case "default":
+			privileges.Seccomp = &swarm.SeccompOpts{
+				Mode: swarm.SeccompModeDefault,
+			}
+		case "unconfined":
+			privileges.Seccomp = &swarm.SeccompOpts{
+				Mode: swarm.SeccompModeUnconfined,
+			}
+		default:
+			// TODO(dperny): is it safe/secure to unconditionally read a file like
+			// this? what if the file is REALLY BIG? is that a user problem for
+			// passing a too-big file, or an us problem for ingesting it
+			// unquestioningly?
+			data, err := os.ReadFile(options.seccomp)
+			if err != nil {
+				// TODO(dperny): return this, or return "unrecognized option" or some such?
+				return nil, errors.Wrap(err, "unable to read seccomp custom profile file")
```

Comment:
Return this. The CLI understands the option and what the user requested. We should tell the user why reading the file failed so they can take corrective action: fix the typo in the file name, sort out file permissions, etc.